### PR TITLE
Iterate on 'Insert link' interface

### LINKED ISSUE #4572: Iterations on link interface
Right now we have a few issues with the link interface that I want to try and solve.

- The links when long can be hidden and hard to read.
- What is a link and when you're interacting with a link is hard to see.
- It feels a complicated process as you're working through.

## First screen: adding a link.
This design has the following points:

- A background to link styling (important to note this could be underline) and would denote in text what is a link. It's important to notice.
- Background on the icon state in editor toolbar. Another visual indicator of the 'adding link state'.
- Ellipsis menu icon to show 'more options', we use this icon already in Gutenberg. This probably is a good debate point, what icon should indicate this?

link-flow-v2

## Second screen: a link added
Once a link is added the following points show:

- Unlink icon swaps for link as you can then unlink. In this mockup you unlink from toolbar.
- The link box fits the url you have added.

link-flow-v2b

## Third screen: link options

This is pretty much as before but with the other changes still applied. If the open in new window ever went from core we could remove it here.

link-flow-v2c

## Fourth screen: Suggested links

This is important to have a little more visual consistency than we have now. It's a small iteration.

link-flow-v2d

## Fifth screen: Auto adjusting width

Here is the auto adjusting in play. If on a larger screen it can stretch to full length.

link-flow-v2e

It's worth noting this design is more of an iteration over redesign. I am aware there will be things we need to discuss here, lets consider that and discuss.

### PR DESCRIPTION
**Pull request by noisysocks**
Closes #4572.

linky-links

- Restores the _Open in new window_ setting by reverting #4583.
- Removes the _Unlink_ button. Instead, links can be removed by toggling off the _Link_ button in the formatting toolbar.
- Styles the _Link Settings_ button to look like the mockup in https://github.com/WordPress/gutenberg/issues/4572.
- Styles the suggested links dropdown to look like the mockup in https://github.com/WordPress/gutenberg/issues/4572.
- Makes it so that the UI expands to fit long URLs when not in editing mode.
- Keeps the _Edit link_ and _Apply_ buttons because I agree with https://github.com/WordPress/gutenberg/issues/4572#issuecomment-360538437.
- Does **not** highlight the selected text while the user is inserting a link. I couldn't make it work—see https://github.com/WordPress/gutenberg/pull/4909#issuecomment-364360350 😢 

#### Testing

1. Create a paragraph block.
2. Highlight some text and click the _Link_ button in the formatting toolbar.
3. Create a link by searching for a page.
4. Create a link by typing in an URL.
5. Toggle on/off _Open in new window_ and check that the generated markup receives a `target="_blank"`.
7. Select an existing link and edit it by clicking on the _Edit_ button in the popup.
6. Select an existing link and remove it by clicking on the _Unlink_ button in the formatting toolbar.

### CODE REVIEW

File: blocks/rich-text/format-toolbar/index.js [hunk 1]
```diff
@@ -124,70 +145,101 @@ class FormatToolbar extends Component {
...
 		const toolbarControls = FORMATTING_CONTROLS.concat( customControls )
 			.filter( control => enabledControls.indexOf( control.format ) !== -1 )
 			.map( ( control ) => {
-				const isLink = control.format === 'link';
+				if ( control.format === 'link' ) {
+					const isFormatActive = this.isFormatActive( 'link' );
+					const isActive = isFormatActive || isAddingLink;
+					return {
+						...control,
+						icon: isFormatActive ? 'editor-unlink' : 'admin-links', // TODO: Need proper unlink icon
```
Comment (noisysocks) [inline blocks/rich-text/format-toolbar/index.js]: I've asked @karmatosed for this icon 😏

**Comment by iseulde** [comment]
> Restores the Open in new window setting by reverting #4583. 

😭 but ok. Wouldn't the ellipsis be better placed on the other side? Similar to blocks, and doesn't deserve first focus imho.

**Comment by karmatosed** [comment]
> Wouldn't the ellipsis be better placed on the other side? Similar to blocks, and doesn't deserve first focus imho.

I do like it before but not oppose to having secondary after edit, it does feel more the same as our current patterns so lets do that change @noisysocks.

Can the link icon be only at full opacity once then text is selected?  This could fix some issues with people just having no text selected see: #5960

**Comment by noisysocks** [comment]
>I do like it before but not oppose to having secondary after edit, it does feel more the same as our current patterns so lets do that change @noisysocks.

👌 done in 0bb6c854d92a78ddb3b1d7a52f3a7493f1f5600c.

> Can the link icon be only at full opacity once then text is selected? This could fix some issues with people just having no text selected see: #5960

Let's treat this as a seperate issue since #5960 is open and I have some questions about it (namely https://github.com/WordPress/gutenberg/issues/5960#issuecomment-379604016).

**Review by karmatosed — APPROVED** [review]
Looks great to me, lets get this in and take from there any iterations, I agree.

**Comment by karmatosed** [comment]
@noisysocks we seem to have a fun bug :) Dancing link interface:



Happens for me in Chrome and Firefox. Props to @mtias for finding this one.

**Pull request merged**